mlflow/tracking/fluent.py

```diff
@@ -3049,3 +3050,74 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark", overwrite=True)
         if "pyspark.ml" in target_library_and_module:
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
+
+
+_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+
+
+class ActiveModel(LoggedModel):
+    """
+    Wrapper around :py:class:`mlflow.entities.LoggedModel` to enable using Python ``with`` syntax.
+    """
+
+    def __init__(self, logged_model: LoggedModel):
+        super().__init__(**logged_model.to_dictionary())
+        self.last_active_model_id = _ACTIVE_MODEL_ID.get()
+        _ACTIVE_MODEL_ID.set(self.model_id)
+        _logger.info(f"Active model set to model with ID: {self.model_id}")
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        _ACTIVE_MODEL_ID.set(self.last_active_model_id)
+
+
+def set_active_model(*, name: Optional[str] = None, model_id: Optional[str] = None) -> ActiveModel:
+    if name is None and model_id is None:
+        raise MlflowException.invalid_parameter_value(
+            message="Either name or model_id must be provided",
+        )
+
+    if model_id is not None:
+        try:
+            logged_model = mlflow.get_logged_model(model_id)
+        except Exception as e:
+            raise MlflowException(f"Failed to find LoggedModel with model_id {model_id}") from e
+        if name is not None and logged_model.name != name:
+            raise MlflowException.invalid_parameter_value(
+                f"LoggedModel with model_id {model_id} has name {logged_model.name}, which does "
+                f"not match the provided name {name}."
+            )
+    elif name is not None:
+        logged_models = mlflow.search_logged_models(
+            filter_string=f"name='{name}'", output_format="list"
+        )
+        if len(logged_models) > 1:
+            _logger.warning(
+                f"Multiple LoggedModels found with name {name}, setting the latest one as active "
+                "model."
+            )
+        if len(logged_models) == 0:
+            _logger.info(f"LoggedModel with name {name} does not exist, creating one...")
```

Comment:
```suggestion
            _logger.info(f"LoggedModel with name {name!r} does not exist, creating one...")
```

Reply:
If `name` is like `hello world`, the message would look like `LoggedModel with name hello world does`, which makes us wonder "is the name hello?". `!r` adds single quotes.